evals/elsuite/ecn.py

```diff
@@ -0,0 +1,102 @@
+import random
+import re
+
+import evals
+import evals.metrics
+from evals import CompletionFn
+
+
+class Ecn(evals.Eval):
+    def __init__(
+        self,
+        completion_fns: list[CompletionFn],
+        samples_jsonl: str,
+        num_few_shot: int = 0,
+        few_shot_jsonl: str = None,
+        **kwargs
+    ):
+        super().__init__(completion_fns, **kwargs)
+        self.samples_jsonl = samples_jsonl
+        self.num_few_shot = num_few_shot
+        self.few_shot_jsonl = few_shot_jsonl
+
+        if self.num_few_shot > 0:
+            assert few_shot_jsonl is not None, "few shot requires few shot sample dataset"
+            self.few_shot_jsonl = few_shot_jsonl
+            self.few_shot = evals.get_jsonl(self.few_shot_jsonl)
+
+    def run(self, recorder):
+        """
+        Called by the `oaieval` CLI to run the eval. The `eval_all_samples` method calls `eval_sample`.
+        """
+
+        samples = self.get_samples()
+        self.eval_all_samples(recorder, samples)
+        events = recorder.get_events("match")
+        num_events = len(list(events))
+
+        total_score = sum(map(lambda e: e.data["score"], events))
+        average_score = total_score / num_events
+        valid_format = f"{round(sum(map(lambda e: e.data['valid_format'], events)) / num_events * 100, 2)}%"
+
+        return {
+            "accuracy": evals.metrics.get_accuracy(events),
+            "total_score": total_score,
+            "average_score": average_score,
+            "valid_format": valid_format,
+        }
+
+    def eval_sample(self, sample, rng: random.Random):
+        """
+        Called by the `eval_all_samples` method to evaluate a single sample.
+
+        ARGS
+        ====
+        `test_sample`: a line from the JSONL test file
+        `rng`: should be used for any randomness that is needed during evaluation
+
+        This method does the following:
+        1. Generate a prompt that contains the task statement, a few examples, and the test question.
+        2. Generate a completion from the model.
+        2. Check if the generated answer is correct.
+        """
+
+        prompt = sample["input"]
+        expected = sample["ideal"].strip()
+        if self.num_few_shot > 0:
```

Comment:
Have we tried non-few-shots? Wondering because I see this line, but it might just be an artefact of copy-pasting the code you started from.

Reply:
indeed, this is a copy-paste artifact, but I didn't do a run without few-shot.